adapter/memquota2/memquota.go

```diff
@@ -48,11 +48,53 @@ type handler struct {
 	windows map[string]*rollingWindow
 
 	// the limits we know about
-	limits map[string]config.Params_Quota
+	limits map[string]*config.Params_Quota
+}
+
+// Limit is implemented by Quota and Override messages.
+type Limit interface {
+	GetMaxAmount() int64
+	GetValidDuration() time.Duration
+}
+
+// matchDimensions matches configured dimensions with dimensions of the instance.
+func matchDimensions(cfg map[string]string, inst map[string]interface{}) bool {
+	for k, val := range cfg {
+		rval := inst[k]
+		if rval == val { // this dimension matches, on to next comparison.
+			continue
+		}
+
+		// if rval has a string representation then compare it with val
+		// For example net.ip has a useful string representation.
+		switch v := rval.(type) {
+		case fmt.Stringer:
+			if v.String() == val {
+				continue
+			}
+		}
+		// rval does not match val.
+		return false
+	}
+	return true
+}
+
+// limit returns the limit associated with this particular request.
+// Check if the instance matches an override, else return the default limit.
+func limit(cfg *config.Params_Quota, instance *quota.Instance) Limit {
+	for idx := range cfg.Overrides {
+		o := cfg.Overrides[idx]
+		if matchDimensions(o.Dimensions, instance.Dimensions) {
```

Comment:
don't the `MakeKey()` methods provide a set of functionality we could use instead of `matchDimensions`?  Is there a reason not to use the existing Key funcs?

Reply:
`MakeKey()` creates a unique key using __all__ dimensions. 
`matchDimensions` needs to match on partial dimensions.